### DISCUSSION TITLE
How to get orientation feedback for crazyflie?

### DISCUSSION THREAD

**Discussion by animesh-singhal** [USC-ACTLab/crazyswarm#607] [Q&A] [answered]
Hi, 

I want to implement a controller to control yaw. I require the yaw feedback to do this. 

While searching for it in the Python API Reference in the [documentation](https://crazyswarm.readthedocs.io/en/latest/index.html), I could only find a function to get the position feedback: [position()](https://crazyswarm.readthedocs.io/en/latest/api.html#pycrazyswarm.crazyflie.Crazyflie.position).

How can I get the feedback for the rest of states? (eg: yaw, roll, pitch). 
 
 Regards, 
 Animesh

**Comment by whoenig** [comment, marked as answer]
It depends on what localization system you are using. Internally, the position() call uses `/tf`, which also includes orientation. However, this value will only be valid, if you use a motion capture system with full pose tracking. For single marker tracking, LightHouse, etc., you would enable pose logging and then subscribe to the the resulting `/pose` topic.

**Reply by animesh-singhal** [reply]
Thank you so much for the prompt response :)
